### PR TITLE
Fix ride_shop_connected dereferencing null.

### PR DESCRIPTION
**Pull request by ZehMatt**
I did not find the issue on the tracker the closest was #8318 which seems to be the same problem. Also given that I'm not sure how to reproduce this error and which realm is affected should I still bump the network version to be safe?

### CODE REVIEW

**Comment by IntelOrca** [comment]
You also need to return if no track element is found. At the moment it just falls through.

**Comment by ZehMatt** [comment]
Not sure I follow, where should I return? I'm checking if the tileElement is null at the beginning and post-loop I validate if the specific element is the track piece and is associated with the ride if not it returns.

File: src/openrct2/ride/Ride.cpp [hunk 1]
```diff
@@ -3472,13 +3471,24 @@ static void ride_shop_connected(Ride* ride)
...
-    tileElement = map_get_first_element_at(x, y);
+    TileElement* tileElement = map_get_first_element_at(x, y);
+    if (tileElement == nullptr)
+    {
+        return;
+    }
+
     do
     {
         if (tileElement->GetType() == TILE_ELEMENT_TYPE_TRACK && tileElement->AsTrack()->GetRideIndex() == ride->id)
```
Comment (duncanspumpkin) [inline src/openrct2/ride/Ride.cpp]: Break this up into two if statements much easier to read. I would introduce a bool found and then check that after the loop

Comment (ZehMatt) [reply src/openrct2/ride/Ride.cpp]: Ha the good old trick from the game actions huh?

Comment (duncanspumpkin) [reply src/openrct2/ride/Ride.cpp]: Yeah

Comment (ZehMatt) [reply src/openrct2/ride/Ride.cpp]: I did it slightly different which avoids two further calls to "AsTrack()" by simply having a TrackElement pointer being assigned or not.

**Pull request merged**